statistics/handle/handle.go

```diff
@@ -1991,3 +2052,40 @@ func (h *Handle) CheckHistoricalStatsEnable() (enable bool, err error) {
 	}
 	return variable.TiDBOptOn(val), nil
 }
+
+func (h *Handle) recordHistoricalStatsMeta(tableID int64, version uint64) error {
+	historicalStatsEnabled, err := h.CheckHistoricalStatsEnable()
+	if err != nil {
+		return errors.Errorf("check tidb_enable_historical_stats failed: %v", err)
+	}
+	if !historicalStatsEnabled {
+		return nil
+	}
+
+	ctx := context.Background()
+	rows, _, err := h.execRestrictedSQL(ctx, "select modify_count, count from mysql.stats_meta where table_id = %? and version = %?", tableID, version)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if len(rows) == 0 {
+		return errors.New("no historical meta stats can be recorded")
+	}
+	modifyCount, count := rows[0].GetInt64(0), rows[0].GetInt64(1)
+
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	exec := h.mu.ctx.(sqlexec.SQLExecutor)
+	_, err = exec.ExecuteInternal(ctx, "begin pessimistic")
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer func() {
+		err = finishTransaction(ctx, exec, err)
+	}()
+
+	const sql = "INSERT INTO mysql.stats_meta_history(table_id, modify_count, count, version, create_time) VALUES (%?, %?, %?, %?, NOW())"
```

Comment:
I suggest that we use `REPLACE INTO` here instead of `INSERT INTO` to make this operation reenterable.

Reply:
Making it reenterable is for safety since the outside caller may not change the `stats_meta`, so if use `INSERT INTO` here, we'll meet a duplicated-key error.

Reply:
> Making it reenterable is for safety since the outside caller may not change the `stats_meta`, so if use `INSERT INTO` here, we'll meet a duplicated-key error.

Done